force-app/main/default/classes/SOQL.cls

```diff
@@ -345,17 +524,28 @@ public with sharing class SOQL {
      */
     public static void setPrimaryContact() {
         //Use the getOpportunityDetailsForClosedWon() method to get the Opportunities
-
+        List<Opportunity> opps = getOpportunityDetailsForClosedWon();
         // Create a set of Account Ids from the Opportunities
-
+        Set<Id> accountIds = new Set<Id>();
+        for (Opportunity opp : opps) {
+            accountIds.add(opp.AccountId);
+        }
         // Query the Contacts
-
+        List<Contact> contacts = [SELECT AccountId, Id
+                                    FROM Contact
+                                    WHERE AccountId IN :accountIds];
+        // Create a map of Account Ids to Contacts
+        Map<Id, Id> accountToContactMap = new Map<Id, Id>();
         // Loop through each Contact and add the Account Id and Contact to the map
-        
+        for (Contact c : contacts) {
+            accountToContactMap.put(c.AccountId, c.Id);
+        }
```

Comment:
What happens to your map if there is more than one Contact per Account? With maps, the new value (the current Contact ID in the loop) will replace the previous Contact ID. As a result, how do we know we're capturing the right Contact ID in the map?